java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -1,16 +1,38 @@
 package glide.api;
 
+import glide.ffi.resolvers.RedisValueResolver;
+import glide.managers.BaseCommandResponseResolver;
 import glide.managers.CommandManager;
 import glide.managers.ConnectionManager;
 import java.util.concurrent.ExecutionException;
+import java.util.function.Function;
 import lombok.AllArgsConstructor;
+import response.ResponseOuterClass.Response;
 
-/** Base Client class for Redis */
+/**
+ * Base Client class for Redis
+ *
+ * @param <T> The data return type.
+ */
 @AllArgsConstructor
-public abstract class BaseClient implements AutoCloseable {
+public abstract class BaseClient<T> implements AutoCloseable {
```

Comment:
Why is the type T attached/defined in the client.  Would the type be defined per-request command, and not for the entire client?  I'd like to see how this works for both standalone and cluster commands when the type returned is known (not customCommand).

Reply:
Fixed in 2b056324.